docs/uml/c4_component_machine.puml

```diff
@@ -0,0 +1,37 @@
+@startuml
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+HIDE_STEREOTYPE()
+LAYOUT_TOP_DOWN()
+
+title Component diagram for DevPod with machine providers
+
+Person(user, "DevPod", "CLI or UI")
+System(prov, "DevPod Provider", "aws, gcloud etc.")
+Component(ide, "IDE", "Local IDE")
+Component(ctx, "Context", "Config, working dir etc.")
+Component(env, "Local Env", "Shell variables, git credentials etc.")
+
+System_Boundary(cluster, "Virtual Machine") {
+  System(controlPlane, "DevPod Agent", "Manages the machine")
+  System(dockerd, "Docker daemon", "Manages the containers")
+  Container_Boundary(ws, "Workspace") {
+    Component(runner, "devcontainer", "")
+    Component(agent, "DevPod Container Agent", "")
+    Component(ssh, "SSH server", "")
+    ContainerDb(dir, "Workspace Directory", "Volume", "")
+  }
+}
+
+BiRel(user, prov, " devpod up ...")
+BiRel(prov, controlPlane, "secure provider specific tunnel")
+Rel(controlPlane, dockerd, "docker run ...")
+Rel(dockerd, ws, "")
+Rel(agent, ssh, "")
+Rel(agent, runner, "")
```

Comment:
If you are trying here to influence the layout without creating actual relations, you cen use `Lay_[U/D/L/R)` to make [the layout look how you want](https://github.com/plantuml-stdlib/C4-PlantUML?tab=readme-ov-file#layout-arrange-elements-without-relationships).

Reply:
Ah good to know! here I did want to add a relation :)